arch/x86/include/mmustructs.h

```diff
@@ -129,17 +129,26 @@ struct mmu_region {
  * region_size has to be provided in bytes
  * for read write access = MMU_ENTRY_READ/MMU_ENTRY_WRITE
  * for supervisor/user mode access = MMU_ENTRY_SUPERVISOR/MMU_ENTRY_USER
+ *
+ * Preprocessor indirection layers used to ensure __COUNTER__ is expanded
+ * properly.
  */
 
-#define MMU_BOOT_REGION(addr, region_size, permission_flags)		\
-	static struct mmu_region region_##addr				\
+#define __MMU_BOOT_REGION(id, addr, region_size, permission_flags)	\
+	static struct mmu_region region_##id				\
 	__attribute__((__section__(".mmulist"), used))  =		\
 	{								\
-		.address = addr,					\
-		.size = region_size,					\
+		.address = (u32_t)(addr),				\
+		.size = (u32_t)(region_size),				\
```

Comment:
a lot of the time we are gonna get the size from taking the address of a linker variable. but its not a big deal to make that an explicit cast, I'll remove these.